### PR TITLE
Add more checkbox styles

### PR DESCRIPTION
**Pull request by duxovni**
Add a "kind" param for StyledCheckbox, allowing designers to choose different styles of checkbox as needed.  In addition to the default kind, there's now an "outline" kind with a green checkmark and a transparent fill.  This is used in the device trust view, since the default checkbox style looks too much like the green "verified" shield and it's awkward to have those next to each other.

This PR is motivated by #7004 and touches the overhauled device list view, so I'm currently basing it on that feature branch. I'll switch it to be based on `develop` once #7004 is merged.

Normal kind:
device-view-4-green-check

Outline kind:
20211029_12h35m42s_grim

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://617c7b05e0b1ae9b1ab2d1ae--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/elements/StyledCheckbox.tsx [hunk 1]
```diff
@@ -17,8 +17,10 @@ limitations under the License.
 import React from "react";
 import { randomString } from "matrix-js-sdk/src/randomstring";
 import { replaceableComponent } from "../../../utils/replaceableComponent";
+import classnames from 'classnames';
 
 interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+    kind?: string;
```
Comment (SimonBrandner) [inline src/components/views/elements/StyledCheckbox.tsx:23]: An enum, please, or at least a union type

Comment (turt2live) [reply src/components/views/elements/StyledCheckbox.tsx:23]: enum would be better, I think.

**Review by turt2live — COMMENTED** [review]
otherwise lgtm

File: src/components/views/elements/StyledCheckbox.tsx (hunk 1 above)
Comment (duxovni) [reply src/components/views/elements/StyledCheckbox.tsx:23]: Sounds good, will do. Will we want to do the same for `AccessibleButton` at some point?

Comment (turt2live) [reply src/components/views/elements/StyledCheckbox.tsx]: Yea, we should.

**Review by turt2live — APPROVED** [review]

**Pull request merged**